File: src/main/Scala/com/bjsxt/scalaspark/core/actions/Action_foreach.scala
package com.bjsxt.scalaspark.core.actions

import org.apache.spark.{SparkConf, SparkContext}

/**
  * foreach 遍历RDD中的每个元素
  */
object Action_foreach {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("foreach").setMaster("local")
    val sc = new SparkContext(conf)
    val lines = sc.textFile("./data/words")
    lines.foreach(println)

  }
}

File: src/main/java/com/bjsxt/javaspark/core/actions/Action_foreach.java
package com.bjsxt.javaspark.core.actions;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class Action_foreach {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setMaster("local").setAppName("foreach");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> lines = sc.textFile("./data/words");
//        lines.foreach(new VoidFunction<String>() {
//            public void call(String s) throws Exception {
//                System.out.println("line is :"+s);
//            }
//        });
        lines.foreach((line)->System.out.println("line is :"+line));

    }
}
